### PR TITLE
[router][producer] A couple of fixes related to SSL for online producer

### PR DESCRIPTION
**Pull request by nisargthakkar**
## A couple of fixes related to SSL for online producer

1. Make routers return if the Kafka cluster is SSL enabled in request_topic call
2. Set Kafka `security.protocol` config for online producer
3. Deprecated `ssl.to.kakfa` due to the misspelling and added a config `kafka.over.ssl` to replace it

## How was this PR tested?

Manual tested so far

## Does this PR introduce any user-facing changes?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/venice-producer/src/main/java/com/linkedin/venice/producer/AbstractVeniceProducer.java [hunk 1]
```diff
@@ -412,9 +414,11 @@ private <D> CompletableFuture<D> getFutureCompletedExceptionally(String exceptio
   @Override
   public void close() throws IOException {
     closed = true;
-    producerExecutor.shutdownNow();
+    producerExecutor.shutdown();
```
Comment (sushantmane) [inline clients/venice-producer/src/main/java/com/linkedin/venice/producer/AbstractVeniceProducer.java:417]: Isn't stopping the immediate stopping of online producers preferred?

Comment (nisargthakkar) [reply clients/venice-producer/src/main/java/com/linkedin/venice/producer/AbstractVeniceProducer.java:417]: Personally, I don't know what is the right way to shutdown an `ExecutorService` the right way. I found this link: https://www.baeldung.com/java-executor-service-tutorial#:~:text=To%20properly%20shut%20down%20an,()%20and%20shutdownNow()%20APIs.&text=This%20method%20returns%20a%20list,to%20do%20with%20these%20tasks. and followed that

Comment (nisargthakkar) [reply clients/venice-producer/src/main/java/com/linkedin/venice/producer/AbstractVeniceProducer.java:417]: By setting `closed = true`, we also then make sure that no new requests are accepted

Comment (nisargthakkar) [reply clients/venice-producer/src/main/java/com/linkedin/venice/producer/AbstractVeniceProducer.java:417]: By setting `closed = true`, we also then make sure that no new requests are accepted

**Review by sushantmane — APPROVED** [review]
The comments are optional. Feel free to drop them. Thanks!

File: clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java [hunk 2]
```diff
@@ -133,10 +133,9 @@ private RouterBackedSchemaReader(
     this.readerSchema = readerSchema;
     this.preferredSchemaFilter = preferredSchemaFilter.orElse(schema -> false);
     readerSchema.ifPresent(AvroSchemaUtils::validateAvroSchemaStr);
-    this.valueSchemaRefreshPeriod = valueSchemaRefreshPeriod;
     this.icProvider = icProvider;
 
-    this.refreshSchemaExecutor = Executors.newSingleThreadScheduledExecutor();
+    this.refreshSchemaExecutor = Executors.newSingleThreadScheduledExecutor(new DaemonThreadFactory("schema-refresh"));
     schemaRefreshFuture = refreshSchemaExecutor.scheduleAtFixedRate(
```
[the remark this answers is not in the document]
Comment (nisargthakkar) [reply clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java:139]: The client has no way of knowing when schemas have changed. The reason for the `ExecutorService` here is that for `asyncUpdate` in online producer, we wish to use the latest superset schema. Hence, for this, we need to periodically refresh schemas. Currently, the default period for the executor is `Integer.MAX_VALUE`, so, it won't eagerly refresh schemas

File: clients/venice-producer/src/main/java/com/linkedin/venice/producer/AbstractVeniceProducer.java (hunk 1 above)
Comment (nisargthakkar) [reply clients/venice-producer/src/main/java/com/linkedin/venice/producer/AbstractVeniceProducer.java]: Reverted the change. Thanks for the discussion

File: clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sushantmane) [reply clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java:139]: [Not related to your PR] I think we should consider building a `watch` functionality for schema changes if there are benefits to using updated value schema.

**Review by sushantmane — APPROVED** [review]
Thanks, @nisargthakkar!

**Pull request merged**